src/pybind/mgr/dashboard/controllers/pool.py

```diff
@@ -140,6 +143,23 @@ def reset_arg(arg, value):
                 reset_arg(arg, '0')
             reset_arg('compression_algorithm', 'unset')
 
+    def _wait_for_pgs(self, pool):
+        current_pool = self._get(pool)
+        initial_pgs = int(current_pool['pg_placement_num']) + int(current_pool['pg_num'])
+        self._pg_wait_loop(current_pool, initial_pgs)
+
+    def _pg_wait_loop(self, pool, initial_pgs):
+        if 'pg_num_target' in pool:
+            target = int(pool['pg_num_target']) + int(pool['pg_placement_num_target'])
+            current = int(pool['pg_placement_num']) + int(pool['pg_num'])
+            if current != target:
+                max_diff = abs(target - initial_pgs)
+                diff = max_diff - abs(target - current)
+                percentage = int(round(diff / float(max_diff) * 100))
+                TaskManager.current_task().set_progress(percentage)
+                time.sleep(4)
```

Comment:
Why are you using 4 seconds here? Is there a special reason?

Reply:
The polling time of the frontend is 5 seconds so we only need to recalculate every 4 seconds as discussed with @tspmelo in his [comment](https://github.com/ceph/ceph/pull/28006#discussion_r284353076)

Reply:
Can you add this as a comment, otherwise this info gets lost and is only visible in the PR.

Reply:
done :)